packages/libs/error-stack/examples/demo.rs

```diff
@@ -49,21 +68,42 @@ fn start_experiments(
                     .attach_printable(format!("Experiment {exp_id} has no valid description"))
             })?;
 
-            let experiment = parse_experiment(description)
+            let experiments = parse_experiment(description)
                 .attach_printable(format!("Experiment {exp_id} could not be parsed"))
                 .change_context(ExperimentError)?;
 
-            Ok(move || experiment.0 * experiment.1)
+            let experiments = experiments
+                .into_iter()
+                .map(|(a, b)| move || a * b)
+                .collect::<Vec<_>>();
+
+            Ok(experiments)
         })
-        .collect::<Result<Vec<_>, ExperimentError>>()
+        .fold(
+            Ok(vec![]),
+            |acc: Result<_, ExperimentError>, value| match (acc, value) {
+                (Ok(mut acc), Ok(value)) => Ok({
+                    acc.extend(value);
+
+                    acc
+                }),
+                (Ok(_), Err(err)) => Err(err),
+                (Err(acc), Ok(_)) => Err(acc),
+                (Err(mut acc), Err(err)) => Err({
+                    acc.extend_one(err);
+
+                    acc
+                }),
+            },
```

Comment:
Same as above

Reply:
fixed in [d1cf758](https://github.com/hashintel/hash/pull/1101/commits/d1cf758932175d95247ed717ca86bc19d0f628ac)